apps/prairielearn/src/ee/lib/lti13.ts

```diff
@@ -596,15 +612,23 @@ class Lti13ContextMembership {
     }
 
     const token = await getAccessToken(lti13_instance.id);
-    const ltiMemberships = await fetchRetry(lti13_course_instance.context_memberships_url, {
+    const fetchArray = await fetchRetry(lti13_course_instance.context_memberships_url, {
       method: 'GET',
       headers: {
         Authorization: `Bearer ${token}`,
         'Content-type': 'application/vnd.ims.lti-nrps.v2.membershipcontainer+json',
       },
     });
 
-    const filteredMemberships = ltiMemberships.members.filter((member: ContextMembership) => {
+    const containers = ContextMembershipContainerSchema.array().parse(fetchArray);
+
+    // Collect the members from the membership container paginated fetch responses
+    // https://www.imsglobal.org/spec/lti-nrps/v2p0/#membership-container-media-type
+    const ltiMemberships = containers.reduce((acc, container) => {
+      return acc.concat(container.members);
+    }, [] as ContextMembership[]);
```

Comment:
I think [`flatMap`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/flatMap) would be a better option here.

Reply:
I'll look at flatMap, thanks.